### ISSUE TITLE
[Question] Can we await specific types of requests via page.route?

### ISSUE THREAD

**Issue by msmith1114** [microsoft/playwright#23680]
As stated in the documentation we can use page.route to await requests via a URL:

```
await page.route('/api/**', route => {
  if (route.request().postData().includes('my-string'))
    route.fulfill({ body: 'mocked-data' });
  else
    route.continue();
});
```

However is it possible to await a certain type of request? IE: I only want to wait on PUT requests to a URL and not a POST request for instance? I looked around and didn't see anything that said we could do this, so I figured i'd ask.

**Comment by darkredstudio** [comment]
https://playwright.dev/docs/api/class-route#route-fallback

```
// Handle POST requests.
await page.route('**/*', route => {
  if (route.request().method() !== 'POST') {
    route.fallback();
    return;
  }
  // Handling POST only.
  // ...
});
```

**Comment by yury-s** [comment]
Closing per the response above, feel free to open a new issue if it doesn't work.
